src/SGB_Command_Palettes.md

```diff
@@ -82,3 +82,20 @@ The palette data is sent by VRAM-Transfer (4 KBytes).
 
 Each Palette consists of four 16-bit color definitions (8 bytes). Note:
 The data is stored at 3000h-3FFFh in SNES memory.
+
+## SGB Command $19 — PAL_PRI
+
+This command can set a flag that will let the game clear and prevent use of 
+the user's SGB palette choice, when it sends 1 of the above commands, except 
+PAL_TRN.
```

Comment:
```suggestion
If the player overrides the active palette set (a pre-defined or the custom one), it stays in effect until the smiley face is selected again, or the player presses the X button on their SNES controller.

However, if `PAL_PRI` is enabled, then changing the palette set (via any of the above commands except `PAL_TRN`) will switch back to the game's newly-modified palette set, if it wasn't already active.
```
The check if for the flag to be zero or not, right?

Also, can you reach a hybrid state e.g. by using all four palettes, and then only sending e.g. a `PAL01` packet? Or would palettes 2 and 3 also revert to the game-defined ones?

Reply:
I like it. I will check the hybrid state later

Reply:
The check is for 0 yes, but due to the `and #$01` mask, it's really `bit 0 set: system pals priority`. I'll change that

The behavior observed (in emulator and bios code, I'll need to write a test rom for hardware) is that there is a separate palette buffer area for system-defined palettes and user-defined palettes:
```
System: $000400-$0005ff (shadow CGRAM)
User: $7ecc30-$7ecc37 (4 colors copied to CGDATA for all of BG0-BG3 palettes)
```
If PAL_PRI is in effect, PAL commands will trigger an update from the system-defined palettes, so palettes 2 and 3 will revert to last palettes set for them (or SGB's default BG palettes). So no hybrid state.

If PAL_PRI is not in effect, they will still be saved, but will only reveal themselves if, for example, the user clicks the 'face' thing that clears the user's palette choice, or if they send PAL_PRI with system priority, then a PAL command after.

Reply:
Definitely, just keeping it thorough :) Yes, the other bits are completely cleared before saving to wram